src/core/symbology-ng/qgssymbolv2.h

```diff
@@ -280,10 +281,26 @@ class CORE_EXPORT QgsMarkerSymbolV2 : public QgsSymbolV2
     QgsMarkerSymbolV2( QgsSymbolLayerV2List layers = QgsSymbolLayerV2List() );
 
     void setAngle( double angle );
-    double angle();
+    double angle() const;
+
+    /** Set "en masse" expression for angle
+     */
+    void setAngleExpression( const QString & exprStr );
+    /** Get "en masse" expression for angle
+     * @return empty if not an expession at the marker level
+     */
+    QString angleExpression() const;
 
     void setSize( double size );
-    double size();
+    double size() const;
+
+    /** Set "en masse" expression for size
+     */
+    void setSizeExpression( const QString & exprStr );
```

Comment:
Hi @nyalldawson, 

I don't mind changing the API, but the underlying storage is expressions, so it won't fix the problem right away. Or do you suggest I change mDataDefinedProperties to be a map of QgsDataDefined ? Because that's a huge change.

As a side note, I think that some parts of QgsDataDefined are necessary (like activate/deactivate) but some parts should be handled by expression, like deciding if it's a simple field or imposing the return type (improvement of qgsexpression required).